File: Code/DevSita/SIta/SIta.Web/Modules/Administration/AdministrationPermissionKeys.cs

using Serenity.Extensibility;
using System.ComponentModel;

namespace Sita.Administration
{
    [NestedPermissionKeys]
    [DisplayName("Administration")]
    public class PermissionKeys
    {
        [Description("User, Role Management and Permissions")]
        public const string Security = "Administration:Security";

        [Description("Languages and Translations")]
        public const string Translation = "Administration:Translation";
        [DisplayName("Bags")]
        public class Bags
        {
            [ImplicitPermission(General), ImplicitPermission(View)]
            public const string Delete = "Sita:dbo.tblBags:Delete";
            [Description("Create/Update"), ImplicitPermission(General), ImplicitPermission(View)]
            public const string Modify = "Sita:dbo.tblBags:Modify";
            public const string View = "Sita:dbo.tblBags:View";

        }
        [DisplayName("Flight")]
        public class Flight
        {
            [ImplicitPermission(General), ImplicitPermission(View)]
            public const string Delete = "Sita:dbo.tblFlight:Delete";
            [Description("Create/Update"), ImplicitPermission(General), ImplicitPermission(View)]
            public const string Modify = "Sita:dbo.tblFlight:Modify";
            public const string View = "Sita:dbo.tblFlight:View";

        }
        [DisplayName("BagsHistory")]
        public class BagsHistory
        {
            [ImplicitPermission(General), ImplicitPermission(View)]
            public const string Delete = "Sita:dbo.tblBagsHistory:Delete";
            [Description("Create/Update"), ImplicitPermission(General), ImplicitPermission(View)]
            public const string Modify = "Sita:dbo.tblBagsHistory:Modify";
            public const string View = "Sita:dbo.tblBagsHistory:View";

        }
        [DisplayName("Field")]
        public class Field
        {
            [ImplicitPermission(General), ImplicitPermission(View)]
            public const string Delete = "Sita:dbo.tblField:Delete";
            [Description("Create/Update"), ImplicitPermission(General), ImplicitPermission(View)]
            public const string Modify = "Sita:dbo.tblField:Modify";
            public const string View = "Sita:dbo.tblField:View";

        }
        [Description("[General]")]
        public const string General = "Sita:General";
    }
}

File: Code/DevSita/SIta/SIta.Web/Imports/ServerTypings/Administration.PermissionKeys.ts
namespace Sita.Administration {
    declare namespace PermissionKeys {
        export const Security = "Administration:Security";
        export const Translation = "Administration:Translation";
        export const General = "Sita:General";

        namespace Bags {
            export const Delete = "Sita:dbo.tblBags:Delete";
            export const Modify = "Sita:dbo.tblBags:Modify";
            export const View = "Sita:dbo.tblBags:View";
        }

        namespace Flight {
            export const Delete = "Sita:dbo.tblFlight:Delete";
            export const Modify = "Sita:dbo.tblFlight:Modify";
            export const View = "Sita:dbo.tblFlight:View";
        }

        namespace BagsHistory {
            export const Delete = "Sita:dbo.tblBagsHistory:Delete";
            export const Modify = "Sita:dbo.tblBagsHistory:Modify";
            export const View = "Sita:dbo.tblBagsHistory:View";
        }

        namespace Field {
            export const Delete = "Sita:dbo.tblField:Delete";
            export const Modify = "Sita:dbo.tblField:Modify";
            export const View = "Sita:dbo.tblField:View";
        }

        namespace SyncData {
            export const Delete = "Sita:dbo.tblConfigSyncData:Delete";
            export const Modify = "Sita:dbo.tblConfigSyncData:Modify";
            export const View = "Sita:dbo.tblConfigSyncData:View";
        }

        namespace CutData {
            export const Delete = "Sita:dbo.TblScheduleCutBag:Delete";
            export const Modify = "Sita:dbo.TblScheduleCutBag:Modify";
            export const View = "Sita:dbo.TblScheduleCutBag:View";
        }
    }
}

